quantum/quantum_keycodes.h

```diff
@@ -671,6 +672,7 @@ enum quantum_keycodes {
 #define MACROTAP(kc) (QK_MACRO | (FUNC_TAP << 8) | (kc))
 #define MACRODOWN(...) (record->event.pressed ? MACRO(__VA_ARGS__) : MACRO_NONE)
 
+#define KC_BSDL BSP_DEL
```

Comment:
Suggest moving this to above `SAFE_RANGE` to preserve VIA compatibility.

Reply:
I'm not sure I understand. All other keycode definitions are in this place, below `SAFE_RANGE`, with the exception of these:

```c
#define _______ KC_TRNS
#define XXXXXXX KC_NO
```
The other definition of `BSP_DEL` is above `SAFE_RANGE`, right next to `GRAVE_ESC`: https://github.com/rpbaptist/qmk_firmware/blob/d46a7da22249be725abb177bae3a217153102432/quantum/quantum_keycodes.h#L119-L120

Reply:
Sorry, I mean, add new keycodes to immediately above `SAFE_RANGE` to preserve the current auto-generated integer values of existing enums.

Reply:
Do you mean like this?

```
diff --git a/quantum/quantum_keycodes.h b/quantum/quantum_keycodes.h
index b0820d303..998249899 100644
--- a/quantum/quantum_keycodes.h
+++ b/quantum/quantum_keycodes.h
@@ -116,7 +116,6 @@ enum quantum_keycodes {
     MAGIC_UNSWAP_ALT_GUI,
     MAGIC_TOGGLE_NKRO,
     MAGIC_TOGGLE_ALT_GUI,
-    BSP_DEL,
     GRAVE_ESC,
 
 // Leader key
@@ -550,6 +549,9 @@ enum quantum_keycodes {
     JS_BUTTON31,
     JS_BUTTON_MAX = JS_BUTTON31,
 
+    // Backspace/Delete functionality
+    BSP_DEL,
+
     // always leave at the end
     SAFE_RANGE
 };
```

Reply:
yes

Reply:
See https://github.com/qmk/qmk_firmware/pull/10661/commits/50ad5bbffaf3bd6b1723e6fd876df3658dafad60